Update cht deployment on GKE in GCP (#1815)
medic/cht-docs: content/en/hosting/4.x/production/kubernetes/gcp-multinode.md

Once mounted, log into your old server, create a session, and run the following rsync command to send data to your new disk. You may have to format the disk in xfs before being able to complete the mount.

> Before running `rsync` command you may need to run the `screen` command first which will allow your `rsync` command to continue running when their window is not visible or if you get disconnected

Reviewer: ```suggestion
Before running `rsync`,  run the `screen` command first which allows the `rsync` command to when you disconnect
```

> 
> ```shell 
>   screen
> ``` 
> or 
> 
> ```shell 
> screen -S <session name>
> ```
> 

Reviewer: ``````suggestion
```shell 
screen
``` 
``````

`rsync -avhWt --no-compress --info=progress2 -e "ssh -i /tmp/identity.pem" /opt/couchdb/data ubuntu@<server_ip>:/<mounted_directory>/`

Run this command from each CouchDB node to a separate storage disk 

### Create namespace

> ```bash
> kubectl create namespace <NAMESPACE_NAME>
> ```

Reviewer: ``````suggestion
```shell
kubectl create namespace <NAMESPACE_NAME>
```
``````

Deploying a service resource allows you to interact with the process that service forwards traffic to over a DNS route simplified to <service_name>.`<namespace>`.svc.cluster.local.

Let's configure some components before deploying service

#### Configmap configuration

Fill out the configmap resource below with the namespace your cht-core project will run in

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: <CONFIGMAP_NAME>
data:
  COUCHDB_SYNC_ADMINS_NODE: couchdb-1.
```

- Apply the secrets configuration:

> ```bash
> kubectl apply -f <COUCHDB_CONFIGMAP_FILE>.yaml
> ```
> 
> - Verify that the  configmap was created successfully:
> 
> ```bash
> kubectl -n <namespace> get configmap
> 
> ```

Reviewer: ``````suggestion
```shell
kubectl apply -f <COUCHDB_CONFIGMAP_FILE>.yaml
```

- Verify that the  configmap was created successfully:

```shell
kubectl -n <namespace> get configmap
```
``````

#### Secrets resources configuration

For the secrets resource, fill out the necessary environment variable values.

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: <SECRET_NAME>
type: Opaque
stringData:
  COUCHDB_PASSWORD: <COUCHDB PASSWORD>
  COUCHDB_SECRET: <COUCHDB SECRET> 
  COUCHDB_USER: <COUCHDB USER>
```

- Apply the secrets configuration:

> ```bash
> kubectl apply -f <COUCHDB_SECRETS_FILE>.yaml
> ```
> 
> - Verify that the secrets were created successfully:
> 
> ```bash
> kubectl -n <namespace> get secret
> 
> ```

Reviewer: ``````suggestion
```shell
kubectl apply -f <COUCHDB_SECRETS_FILE>.yaml
```

- Verify that the secrets were created successfully:

```shell
kubectl -n <namespace> get secret
```
``````

After configuring the storage components, you need to create a deployment for CouchDB that will use the persistent storage. This deployment defines how your CouchDB instance will run within Kubernetes. We will also create configmap and secrets resource to hold our credentials in one location for all templates.

- Apply the secrets configuration:

> ```bash
> kubectl apply -f <COUCHDB_SERVICES_FILE>.yaml
> ```
> 
> - Verify that the  configmap was created successfully:
> 
> ```bash
> kubectl -n <namespace> get services
> 
> ```

Reviewer: ``````suggestion
```shell
kubectl apply -f <COUCHDB_SERVICES_FILE>.yaml
```

- Verify that the  configmap was created successfully:

```shell
kubectl -n <namespace> get services
```
``````

          - containerPort: 5008
```

- Apply the deployment configuration:

> ```bash
> kubectl apply -f <CHT_CORE_DEPLOYMENT_FILE>.yaml
> ```
> 
> - Verify that the deployment was created successfully:
> 
> ```bash
> kubectl get deployments
> kubectl get pods
> ```

Reviewer: ``````suggestion
```shell
kubectl apply -f <CHT_CORE_DEPLOYMENT_FILE>.yaml
```

- Verify that the deployment was created successfully:

```shell
kubectl get deployments
kubectl get pods
```
``````

  type: ClusterIP
```

* Apply the deployment configuration:

> ```bash
>   kubectl apply -f <CHT_SERVICE_FILE.yaml>
> ```
> 
> - Verify that the deployment was created successfully:
> 
> ```bash
>   kubectl get deployments
>   kubectl get pods
> ```

Reviewer: ``````suggestion
```shell
kubectl apply -f <CHT_SERVICE_FILE.yaml>
```

- Verify that the deployment was created successfully:

```shell
kubectl get deployments
kubectl get pods
```
``````

Once the certs are generated, create a [kubernetes secret tls](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_create/kubectl_create_secret_tls/)

```bash
kubectl -n <namespace> create secret tls --cert=/etc/letsencrypt/live/<domain>/fullchain.pem --key=/etc/letsencrypt/live/<domain>/privkey.pem
```

Verify the secret was created by running:

> ```bash
> kubectl -n <namespace> get secrets
> ```

Reviewer: ``````suggestion
```shell
kubectl -n <namespace> get secrets
```
``````
